implementations/rust/ockam/ockam/src/entity/local.rs

```diff
@@ -0,0 +1,219 @@
+use crate::{
+    Contact, ContactsDb, Entity, KeyAttributes, Profile, ProfileAuth, ProfileChangeEvent,
+    ProfileChanges, ProfileContacts, ProfileEventAttributes, ProfileIdentifier, ProfileIdentity,
+    ProfileSecrets, ProfileSync, RemoteEntity, RemoteForwarder, RemoteForwarderInfo, Route,
+    SecureChannelTrait,
+};
+use ockam_core::{Address, AddressSet, Message, Result, Worker};
+use ockam_node::{Cancel, Context};
+use ockam_vault_core::{PublicKey, Secret};
+use ockam_vault_sync_core::Vault;
+
+use ockam_core::hashbrown::HashMap;
+
+pub struct LocalEntity {
+    pub ctx: Context,
+    vault: Address,
+    entity: Entity<ProfileSync>,
+    secure_channels: HashMap<Address, Route>,
+}
+
+impl LocalEntity {
+    pub async fn create<A: Into<Address>>(node_ctx: &Context, address: A) -> Result<LocalEntity> {
+        let ctx = node_ctx.new_context(address).await?;
```

Comment:
Can we just generate random address here?

Reply:
Yes, I suppose when/if they need to be named explicitly we can add it then.